src/main/java/com/metamx/emitter/core/HttpPostEmitter.java

```diff
@@ -125,39 +131,57 @@ public HttpPostEmitter(
     catch (MalformedURLException e) {
       throw new ISE(e, "Bad URL: %s", config.getRecipientBaseUrl());
     }
+    concurrentBatch.set(new Batch(this, acquireBuffer(), 0));
   }
 
   @Override
   @LifecycleStart
   public void start()
   {
-    synchronized (started) {
-      if (!started.getAndSet(true)) {
-        exec.schedule(
-            new EmittingRunnable(version.get()),
-            config.getFlushMillis(),
-            TimeUnit.MILLISECONDS
-        );
+    synchronized (startLock) {
+      if (!started) {
+        if (startLatch.getCount() == 0) {
+          throw new IllegalStateException("Already started.");
+        }
+        started = true;
+        startLatch.countDown();
```

Comment:
`started` flag is checked only in start() and close() methods. The order of `startLatch.countDown()` and `emittingThread.start()` doesn't matter. Threads which call emit(Event) could proceed and add batches to the `buffersToEmit` queue, they don't communicate with the emittingThread at all. The emittingThreads spins on the `buffersToEmit` and if there are no events, just waits.